api/spa.go

```diff
@@ -12,6 +12,8 @@ import (
 
 func SinglePageApplication(ui *ui.Config) fiber.Handler {
 	return func(c *fiber.Ctx) error {
+		theme := string(c.Request().Header.Cookie("theme"))
+		ui.DarkMode = theme == "dark"
```

Comment:
Because this is the field of a pointer, it might change the default `ui.DarkMode` for other users. It might even panic if there's concurrent modifications to the field, though I imagine this would be a rare occurrence. 

I'll push some adjustments on your fork